src/caf/toolkit/log_helpers.py

```diff
@@ -846,3 +848,33 @@ def capture_warnings(
 
     if file_handler_args is not None:
         warning_logger.addHandler(get_file_handler(**file_handler_args))
+
+
+def write_metadata(path: Path, details: ToolDetails, **metadata: Any) -> None:
+    """
+    Write metadata (tool details, system information and any user specified
+    metadata) out to a user specified path.
```

Comment:
Splitting this out so the short description fits onto one line
```suggestion
    Write metadata out to a user specified path.
    
    This includes tool details, system information and any user specified
    metadata.
```